tests/xharness/Process_Extensions.cs

```diff
@@ -56,17 +57,26 @@ public static Task<bool> PollForExitAsync (int pid, TimeSpan timeout)
 
 	public static class Process_Extensions
 	{
-		public static async Task<ProcessExecutionResult> RunAsync (this Process process, Log log, CancellationToken? cancellation_token = null, bool? diagnostics = null)
+		public static async Task<ProcessExecutionResult> RunAsync (this Process process, ILog log, CancellationToken? cancellation_token = null, bool? diagnostics = null)
 		{
 			return await RunAsync (process, log, log, log, cancellation_token: cancellation_token, diagnostics: diagnostics);
 		}
 
-		public static Task<ProcessExecutionResult> RunAsync (this Process process, Log log, bool append = true, TimeSpan? timeout = null, Dictionary<string, string> environment_variables = null, CancellationToken? cancellation_token = null, bool? diagnostics = null)
+		public static Task<ProcessExecutionResult> RunAsync (this Process process, ILog log, bool append = true, TimeSpan? timeout = null, Dictionary<string, string> environment_variables = null, CancellationToken? cancellation_token = null, bool? diagnostics = null)
 		{
 			return RunAsync (process, log, log, log, timeout, environment_variables, cancellation_token, diagnostics);
 		}
 
-		public static async Task<ProcessExecutionResult> RunAsync (this Process process, Log log, TextWriter StdoutStream, TextWriter StderrStream, TimeSpan? timeout = null, Dictionary<string, string> environment_variables = null, CancellationToken? cancellation_token = null, bool? diagnostics = null)
+		public static Task<ProcessExecutionResult> RunAsync (this Process process, ILog log, ILog stdoutLog, ILog stderrLog, TimeSpan? timeout = null, Dictionary<string, string> environment_variables = null, CancellationToken? cancellation_token = null, bool? diagnostics = null)
+		{
+			if (stdoutLog is TextWriter StdoutStream && stderrLog is TextWriter StderrStream) {
+				return RunAsync (process, log, StdoutStream, StderrStream, timeout, environment_variables, cancellation_token, diagnostics);
```

Comment:
I don't understand why we need this cast. Seems like the params for the method below could also be ILog and nothing would break?

Reply:
It will not compile. Needs to be casted. I did check without it.

Reply:
I suggest changing the signature below to `ILog` and remove this one. It probably doesn't compile because the signatures will be the same or is there something inside the method below that needs `TextWriter` and is not ok with `ILog`?

Reply:
Correct, I can remove and either do the casting in the call or in the other method. I added the other one 'cause the indentation went crazy. I'll do it in the caller since if will be cleaner.

Reply:
I am trying to say that I think there doesn't need to be any casting at all :)